src/Core/Checkout/Cart/Event/AdminPromotionCodeRedeemedEvent.php

```diff
@@ -0,0 +1,35 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Checkout\Cart\Event;
+
+use Shopware\Core\Checkout\Order\OrderEntity;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\Log\Package;
+use Symfony\Contracts\EventDispatcher\Event;
+
+#[Package('checkout')]
+class AdminPromotionCodeRedeemedEvent extends Event
+{
+    final public const EVENT_NAME = 'admin.promotion.code.redeemed';
```

Comment:
I don't think we need a name here, this is usually just used for things in the flow builer.